core/integration/module_test.go

```diff
@@ -6770,6 +6770,877 @@ func schemaVersion(ctx context.Context) (string, error) {
+			{
+				sdk: "typescript",
+				source: `import { Directory, File,object, func, argument } from "@dagger.io/dagger"
+@object()
+class Test {
+  @func()
+  async tooHighRelativeDirPath(@argument({ defaultPath: "../../" }) backend: Directory): Promise<string[]> {
+    // The engine should throw an error
+    return []
+  }
+
+  @func()
+	async nonExistingPath(@argument({ defaultPath: "/invalid" }) dir: Directory): Promise<string[]> {
+    // The engine should throw an error
+    return []
+  }
+
+  @func()
+	async tooHighRelativeFilePath(@argument({ defaultPath: "../../file.txt" }) backend: File): Promise<string> {
+    // The engine should throw an error
+    return ""
+  }
+
+	@func() nonExistingFile(@argument({ defaultPath: "/invalid" }) file: File): Promise<string> {
+    // The engine should throw an error
+    return ""
+  }
+}
+`,
+			},
+		} {
+			tc := tc
+
+			t.Run(tc.sdk, func(ctx context.Context, t *testctx.T) {
+				c := connect(ctx, t)
+
+				modGen := goGitBase(t, c).
+					WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+					WithWorkdir("/work").
+					With(daggerExec("init", "--name=test", "--sdk="+tc.sdk, "--source=dagger")).
+					WithWorkdir("/work/dagger").
+					With(sdkSource(tc.sdk, tc.source)).
+					WithWorkdir("/work")
+
+				t.Run("too high relative context dir path", func(ctx context.Context, t *testctx.T) {
+					out, err := modGen.With(daggerCall("too-high-relative-dir-path")).Stdout(ctx)
+					require.Empty(t, out)
+					require.Error(t, err)
+					require.ErrorContains(t, err, `path should be relative to the context directory`)
+				})
+
+				t.Run("too high relative context file path", func(ctx context.Context, t *testctx.T) {
+					out, err := modGen.With(daggerCall("too-high-relative-file-path")).Stdout(ctx)
+					require.Empty(t, out)
+					require.Error(t, err)
+					require.ErrorContains(t, err, `path should be relative to the context directory`)
+				})
+
+				t.Run("non existing dir path", func(ctx context.Context, t *testctx.T) {
+					out, err := modGen.With(daggerCall("non-existing-path")).Stdout(ctx)
+					require.Empty(t, out)
+					require.Error(t, err)
+					require.ErrorContains(t, err, "no such file or directory")
+				})
+
+				t.Run("non existing file", func(ctx context.Context, t *testctx.T) {
+					out, err := modGen.With(daggerCall("non-existing-file")).Stdout(ctx)
+					require.Empty(t, out)
+					require.Error(t, err)
+					require.ErrorContains(t, err, "no such file or directory")
+				})
+			})
+		}
+	})
+
+	t.Run("deps", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		ctr := goGitBase(t, c).
+			WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+			WithWorkdir("/work/dep").
+			With(daggerExec("init", "--source=.", "--name=dep", "--sdk=go")).
+			WithNewFile("main.go", `package main
+
+import (
+	"dagger/dep/internal/dagger"
+)
+
+type Dep struct{}
+
+func (m *Dep) GetSource(
+	// +defaultPath="/dep"
+	// +ignore=["!yo"]
+	source *dagger.Directory,
```

Comment:
nit: I think it's worth adding coverage of relative paths when setting `defaultPath` for deps in addition to this coverage of absolute paths. There's a lot of logic to get correct when it comes to resolving those paths so seems worth another test.

Reply:
Done in [79e917c](https://github.com/dagger/dagger/pull/7744/commits/79e917cb5a27c2baaa07b013b17e1688264ac003), with fixes on relative dep module resolution normally.